Converting IntroToAnnotationDbi.Rnw to Rmarkdown

BerylKanali opened the pull request:
@jwokaty 
Here is a draft pull request. 
The issue is, I need help to figure out what library or a way I can use to automatically section the exercise and solution bit (line 186- 280) like it was done in the Rnw file using the Theorem package:

```
%% Question, Exercise, Solution
\usepackage{theorem}
\theoremstyle{break}
\newtheorem{Ext}{Exercise}
\newtheorem{Question}{Question}

\newenvironment{Exercise}{
  \renewcommand{\labelenumi}{\alph{enumi}.}\begin{Ext}%
}{\end{Ext}}
\newenvironment{Solution}{%
  \noindent\textbf{Solution:}\renewcommand{\labelenumi}{\alph{enumi}.}%
}{\bigskip}
```
If not, I can do it manually.
[PDF](https://bioconductor.org/packages/release/bioc/vignettes/AnnotationDbi/inst/doc/IntroToAnnotationPackages.pdf) file
[HTML](https://rpubs.com/BerylK/1002750) file so far

Note: Replaced the dataTypes pdf image file to png for better output.

BerylKanali committed:
Convert file to Rmd

jwokaty commented:
AFAIK there's no automatic way to convert these and I don't think there is an equivalent in R Markdown. It is possible to mock this up in HTML just using R Markdown to make it look similar to the PDF manually then remove the section, like the one above, since it would no longer be used.

LiNk-NY commented:
You can create a `<details>` html tag in the vignette to hide or show the solution, e.g.,

```r
<details> Solution <summary> Here is the solution </summary>
## R code chunk of the solution
</details>
```

There is also the `learnr` package for `shiny` documents but the infrastructure is more involved and it requires a `shiny` dependency.

BerylKanali committed:
Update Rmd file

BerylKanali committed:
Update DESCRIPTION

BerylKanali committed:
update Rmd contents

BerylKanali committed:
delete Rnw file

BerylKanali committed:
marge branch with remote

BerylKanali commented:
@jwokaty 
Added some commits here and switched it from draft to ready for review.
Updated[ HTML](https://rpubs.com/BerylK/1002750)

sonali8434 commented:
Hello @BerylKanali, I am going to review your PR. We will be using this checklist to review it.

- [x] the `.Rmd` file knits to `HTML`
- [x] `R CMD build` runs without errors or timeouts
- [x] the tarball from `R CMD build` contains the `HTML` (check with the
  following by substituting the package name and vignette name `tar ztf
  package_name.tar.gz | grep 'doc/vignette_name'`)
- [x] the `.Rnw` file has been removed
- In the `DESCRIPTION` file
  - [x] `BiocStyle` and `knitr` are listed in `Suggests`
  - [x] the line `VignetteBuilder: knitr` exists
  - [x] any added lines use the same spacing and indents as the existing document
- If agreed with the Maintainer
  - [x] the contributor is in the author list in the `DESCRIPTION` file*.
  - [x] the contributor is in the author list in the vignette's YAML*.
- If this pull request involves converting from separate `Author` and  `Maintainer`
  lines to the `Authors@R` vector, please ensure that
  - [x] the `Authors@R` vector includes the maintainer as specified with `role='cre'`.
  - [x] the `Maintainer` line is completely removed.
- [x] `HTML` document is representative of the `PDF` in content and
  in general the presentation
- [ ] Where the contributor was not able to preserve the content and presentation
  of the `PDF` is noted as a comment in the pull request
- [x] the R Markdown file is representative of the Sweave document and follows
  best practices, such as replacing links to Bioconductor packages with calls
  to BiocStyle's `Biocpkg()`
- [x] Only files necessary for the conversion are included in the pull request.
- [x] Long lines have been broken up and reformatted, you can achieve this by
  selecting the text in your script and then clicking on `Code > Reflow Comment`
  in R Studio.
- [x] Code blocks are surrounded by a blank line, and there are no blank lines
  before the start or after the end of the code within the code block. 

* Refer to the contribution guide for acceptable formats

sonali8434 commented on DESCRIPTION:
```diff
@@ -8,7 +8,13 @@ BugReports: https://github.com/Bioconductor/AnnotationDbi/issues
+	person("Nianhua", "Li", role="aut"),
+	person("Beryl", "Kanali", role="ctb",
+	  comment="Converted 'IntoToAnnotationPackages' vignette from Sweave to RMarkdown."))
 Maintainer: Bioconductor Package Maintainer <maintainer@bioconductor.org>
```
We have to include maitainer in `Authors@R: c` vector and line 18 should be completly removed.
```
Authors@R: c(
	person("Hervé", "Pagès", role="aut"),
	person("Marc", "Carlson", role="aut"),
	person("Seth", "Falcon", role="aut"),
	person("Nianhua", "Li", role="aut"),
        person("Bioconductor Package Maintainer", role="cre",
          email="maintainer@bioconductor.org"),
	person("Beryl", "Kanali", role="ctb",
	  comment="Converted 'IntoToAnnotationPackages' vignette from Sweave to RMarkdown."))
```

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
````diff
@@ -0,0 +1,383 @@
+knitr::include_graphics("databaseTypes.png")
+```
+
+```{r echo=FALSE, include=FALSE}
+if (!require("BiocManager", quietly = TRUE))
+    install.packages("BiocManager")
+
+BiocManager::install("BiocStyle", force = TRUE)
+```
+
+
+```{r include=FALSE}
+library(knitr)
+opts_chunk$set(tidy=FALSE)
+```
+
+
+```{r include=FALSE}
+library(BiocStyle)
+```
````
We don't need these lines 24-40, I Remember that One of the requirements of a vignette is that it must not install anything without the user's knowledge. Also BiocStyle and knitr are included in suggests.

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
````diff
@@ -0,0 +1,383 @@
+
+```{r  label = fig:dbtypes, echo = FALSE, fig.align="center", fig.cap = "Annotation Packages: the big picture", out.width = '60%'}
+knitr::include_graphics("databaseTypes.png")
+```
````
I think fig is not a valid chunk label, fig.align is by default center so we don't need to mention it here please correct this code chunk label this should be.

````
```{r dbtypes, fig.cap = "Annotation Packages: the big picture", echo = FALSE, fig.width=0.6}
knitr::include_graphics("databaseTypes.png")
```
````

sonali8434 replied:
I think the figure is too small as when we compare it from pdf replace `out.width = '60%'` to `fig.width=0.6`

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
````diff
@@ -0,0 +1,383 @@
+```
+
+*Bioconductor* provides extensive annotation resources. These can be
+*genecentric*, or *genome centric*. Annotations can be provided in packages
````
There should be space between gene and centric here `*gene centric*`

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,383 @@
+*genecentric*, or *genome centric*. Annotations can be provided in packages
+curated by *Bioconductor*, or obtained from web-based resources. This vignette
+is primarily concerned with describing the annotation resources that are
+availableas packages. More advanced users who wish to learn about how to make
```
There should be space between `available` and `as` here

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,383 @@
+curated by *Bioconductor*, or obtained from web-based resources. This vignette
+is primarily concerned with describing the annotation resources that are
+availableas packages. More advanced users who wish to learn about how to make
+newannotation packages should see the vignette titled "Creating select
```
There should be space here between `new` and  `annotation`.

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,383 @@
+is primarily concerned with describing the annotation resources that are
+availableas packages. More advanced users who wish to learn about how to make
+newannotation packages should see the vignette titled "Creating select
+Interfacesfor custom Annotation resources" from the 
```
There should be space between `Interfaces` and `for` Please correct all the instances such as these and let me know I will continue my review after you correct these instances.

sonali8434 reviewed (changes requested):
Hello @BerylKanali Thanks for your PR, during the review process I noticed that there are many instances where we need a space between two words please correct the instances like these and let me know I will continue the review process.

BerylKanali committed:
Update Rmd file

BerylKanali committed:
Update DESCRIPTION

BerylKanali commented:
Hi @sonali8434 I made the changes, here is the updated [HTML](https://rpubs.com/BerylK/1002750) file.

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,366 @@
+available as packages. More advanced users who wish to learn about how to make
+new annotation packages should see the vignette titled "Creating select
+Interfaces for custom Annotation resources" from the `r
+Biocpkg("AnnotationForge")` package.
```
`r Biocpkg("AnnotationForge")` should be in single line to render it correctly in html file

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,366 @@
+-   Generic genome features: Can generate via `r Biocpkg("GenomicFeatures")`
+
+One web-based resource accesses [biomart](http://www.biomart.org/), via the `r
+Biocpkg("biomaRt")` package: 
```
`r Biocpkg("biomaRt")` should be in one single line

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,366 @@
+use of a new set of methods to more easily access their contents. These
+four methods are named: `columns`, `keytypes`, `keys` and `select`. And they are
+described in this vignette. They can currently be used with all chip, organism,
+and `TxDb` packages along with the popular GO.db package.
```
GO.db should be `r Biocpkg("GO.db")`

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,366 @@
+As previously mentioned, a new set of methods have been added that allow a
+simpler way of extracting identifier based annotations. All the annotation
+packages that support these new methods expose an object named exactly the same
+as the package itself. These objects are collectively called `AnnotationDb`
```
I think `AnnotationDb` should be italicized as shown in pdf

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,366 @@
+as the package itself. These objects are collectively called `AnnotationDb`
+objects for the class that they all inherit from. The more specific classes (the
+ones that you will actually see in the wild) have names like `OrgDb`, `ChipDb`
+or `TxDb` objects. These names correspond to the kind of package (and underlying
```
the words `OrgDb`,  `ChipDb` and  `TxDb` should be italicized

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,366 @@
+objects are `columns`, `keys`, `keytypes` and `select`.
+
+In addition, another accessor has recently been added which allows extraction of
+one column at at time. the `mapIds` method allows users to extract data into
```
@jwokaty I think there is a spelling mistake here in a original pdf  there is two `at` but I think there should be `at a`, can we correct this in the .Rmd file?

BerylKanali replied:
I have corrected it TO `at a`. @jwokaty is that okay?

jwokaty replied:
Yes. Thank you. Can we also capitalize the `T` at the beginning of the sentence

    the `mapIds` method

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,366 @@
+In addition, another accessor has recently been added which allows extraction of
+one column at at time. the `mapIds` method allows users to extract data into
+either a named character vector, a list or even a SimpleCharacterList. This
+method should work with all the different kinds of `AnnotationDb` objects
```
`AnnotationDb` should be italicized @BerylKanali please italicized all the objects in this .Rmd file.

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,366 @@
+
+# ChipDb objects and the select method
+
+An extremely common kind of Annotation package is the so called platform based
```
Annotation should be `r Biocpkg("Annotation")`

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,366 @@
+An extremely common kind of Annotation package is the so called platform based
+or chip based package type. This package is intended to make the manufacturer
+labels for a series of probes or probesets to a wide range of gene-based
+features. A package of this kind will load an `ChipDb` object. Below is a set of
```
Please italicized `ChipDb` and all the other objects in this .Rmd file.

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,366 @@
+Database, or `r Biocpkg("eg")` for Entrez Gene ids).
+
+Just as the chip packages load a `ChipDb` object, the org packages will load a
+`OrgDb` object. The following exercise should acquaint you with the use of these
```
Please italicize `ChipDb` and `OrgDb` object.

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,366 @@
+
+**Exercise 1**
+
+Display the `OrgDb` object for the `r Biocpkg("org.Hs.eg.db")` package.
```
Please italicize the `OrgDb` object

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,366 @@
+**Exercise 2**
+
+Please run the following code snippet (which will load a fake data result that I
+have provided for the purposes of illustration):
```
these two lines should be italicize
``` 
*Please run the following code snippet (which will load a fake data result that I have provided for the purposes of illustration):*
```

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,366 @@
+The rownames of this table happen to provide entrez gene identifiers for each
+row (for human). Find the gene symbol and gene name for each of the rows in
+resultTable and then use the merge method to attach those annotations to it.
+
```
these three lines should be italicize and remove the extra blank line 236 
```
*The rownames of this table happen to provide entrez gene identifiers for each
row (for human). Find the gene symbol and gene name for each of the rows in
resultTable and then use the merge method to attach those annotations to it.*

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,366 @@
+resultTable and then use the merge method to attach those annotations to it.
+
+
+**Solution:**
```
add a space (carriage return) between solution and code block

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,366 @@
+
+# Using select with GO.db
+
+When you load the GO.db package, a `GODb` object is also loaded. This allows you
```
Please change this line to 
```
When you load the ` r Biocpkg("GO.db")` package, a *GODb* object is also loaded. This allows you
```

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,366 @@
+
+# Using select with TxDb packages
+
+A `TxDb` package (a 'TxDb' package) connects a set of genomic coordinates to
```
Please change this line to 
```
A `r Biocpkg("TxDb")` package (a 'TxDb' package) connects a set of genomic coordinates to

sonali8434 replied:
Sorry @BerylKanali Please change it to 

```
A `TxDb` package (a 'TxDb' package) connects a set of genomic coordinates to
```

since this link is not working

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,366 @@
+
+**Exercise 3**
+
+Display the `TxDb` object for the `r Biocpkg("TxDb.Hsapiens.UCSC.hg19.knownGene")`
```
Please change this line to 
```
Display the *TxDb* object for the `r Biocpkg("TxDb.Hsapiens.UCSC.hg19.knownGene")`

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
````diff
@@ -0,0 +1,366 @@
+```
+
+As is widely known, in addition to providing access via the `select` method,
+`TxDb` objects also provide access via the more familiar `transcripts`, `exons`,
````
Please italicize `TxDb` Generally we italicize objects.

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,366 @@
+# Using select with EnsDb packages
+
+Similar to the `TxDb` objects/packages discussed in the previous section,
+`EnsDb` objects/packages provide genomic coordinates of gene models along with
```
please italicize `TxDb` and `EnsDb` objects

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,366 @@
+`EnsDb` objects/packages provide genomic coordinates of gene models along with
+additional annotations (e.g. gene names, biotypes etc) but are tailored to
+annotations provided by Ensembl. The central methods `columns`, `keys`,
+`keytypes` and `select` are all implemented for `EnsDb` objects. In addition,
```
please  italicize `EnsDb`

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,366 @@
+additional annotations (e.g. gene names, biotypes etc) but are tailored to
+annotations provided by Ensembl. The central methods `columns`, `keys`,
+`keytypes` and `select` are all implemented for `EnsDb` objects. In addition,
+these methods allow also the use of the `EnsDb` specific filtering framework to
```
please italicize *EnsDb*

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
````diff
@@ -0,0 +1,366 @@
+    install.packages("BiocManager")
+
+BiocManager::install("EnsDb.Hsapiens.v75")
+```
````
Please remove line 316 - 321 we don't have this code block in .Rnw file.

sonali8434 commented on vignettes/IntroToAnnotationPackages.Rmd:
````diff
@@ -0,0 +1,366 @@
+```
+
+We can modify the queries above to retrieve only genes encoded on chromosome Y.
+To this end we use filter objects available for `EnsDb` objects and its methods.
````
Please change this line to 
```
To this end we use filter objects available for *EnsDb* objects and its methods.

sonali8434 reviewed (changes requested):
Hello @Berylkanali Thank you for the changes I have completed the review Please do these modifications after that this PR will be ready for the final review.

I noticed that you have not italicized the objects in many instances please correct that rest everything looks good.

BerylKanali committed:
Update Rmd after review

BerylKanali committed:
update description

Merge branch 'AnnotationDbi-Rmd' of https://github.com/BerylKanali/AnnotationDbi into AnnotationDbi-Rmd
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

BerylKanali committed:
Merge branch 'devel' into AnnotationDbi-Rmd

BerylKanali commented:
@sonali8434 Thank you for your review, I have made the changes. Here is the updated [HTML](https://rpubs.com/BerylK/1002750) file.

sonali8434 commented:
Hello, @BerylKanali Thank you so much for the quick edits; just two small changes then it will be ready for @jwokaty.

sonali8434 commented:
Hello @jwokaty, The review of this file is complete. You can do a final review after these two changes are addressed.
I have checked R CMD build and HTML in the tarball

Thank you!!

jwokaty commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,361 @@
+
+- Organism level: e.g. `r Biocpkg("org.Mm.eg.db")`. 
+- Platform level: e.g. `r Biocpkg("hgu133plus2.db")`, 
+`r Biocpkg("hgu133plus2.probes")`, `r Biocpkg("hgu133plus2.cdf")` 
```
Maybe these were renamed or misspelled in the original document? Biocpkg can't make links for these, but I did find packages `hgu133plus2probe` and `hgu133plus2cdf`. Could you update line 37 with these?

jwokaty commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,361 @@
+
+# ChipDb objects and the select method
+
+An extremely common kind of `r Biocpkg("Annotation")` package is the so called
```
I think the writer is speaking generally about Annotation packages but not a specific package named Annotation. Remove

    `r Biockpg("")`

jwokaty commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,361 @@
+(e.g. Entrez Gene id) and contains mappings between this identifier and other
+kinds of identifiers (e.g. GenBank or Uniprot accession number, RefSeq id,
+etc.). The name of an org package is always of the form org.\<Ab>\.\<id>\.db
+(e.g.`r Biocpkg("org.Sc.sgd.db")`where \<Ab>\ is a 2-letter abbreviation of the
```
Add a closing parenthesis and space between the last backtick and `where` so that we have

    (e.g.`r Biocpkg("org.Sc.sgd.db")`) where

jwokaty commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,361 @@
+(e.g.`r Biocpkg("org.Sc.sgd.db")`where \<Ab>\ is a 2-letter abbreviation of the
+organism (e.g.`r Biocpkg("Sc")` for *Saccharomyces cerevisiae*) and \<id>\ is an
+abbreviation (in lower-case) describing the type of central identifier (e.g.
+`r Biocpkg("sgd")` for gene identifiers assigned by the Saccharomyces Genome
+Database, or `r Biocpkg("eg")` for Entrez Gene ids).
```
`sgd` and `eg` are not packages. Remove

    `r Biocpkg("")`

jwokaty commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,361 @@
+kinds of identifiers (e.g. GenBank or Uniprot accession number, RefSeq id,
+etc.). The name of an org package is always of the form org.\<Ab>\.\<id>\.db
+(e.g.`r Biocpkg("org.Sc.sgd.db")`where \<Ab>\ is a 2-letter abbreviation of the
+organism (e.g.`r Biocpkg("Sc")` for *Saccharomyces cerevisiae*) and \<id>\ is an
```
`Sc` is not a package. Remove

    `r Biocpkg("")`

jwokaty commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,361 @@
+
+# Using select with TxDb packages
+
+A `r Biocpkg("TxDb")` package (a 'TxDb' package) connects a set of genomic
```
`TxDb` is not a package, but a type of package. Remove

    `r Biocpkg("")`

jwokaty commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,361 @@
+  - name: "Marc Carlson"
+  - name: "Beryl Kanali"
+    affiliation: "Converted vignette from Sweave to Rmarkdown"
+date: "Last modified: November 2022; Compiled: `r format(Sys.time(), '%B %d , %Y')`"
```
Remove `Last modified: November 2022; Compiled: `. I don't know where the November 2022 date comes from.

jwokaty commented on vignettes/IntroToAnnotationPackages.Rmd:
```diff
@@ -0,0 +1,361 @@
+
+# Using select with GO.db
+
+When you load the ` r Biocpkg("GO.db")` package, a *GODb* object is also loaded.
```
Remove the space after the first backtick (otherwise it doesn't render the link properly).

jwokaty reviewed (changes requested):
@BerylKanali There are a few more changes. Mostly changes to remove Biocpkg() from things aren't packages. The easiest way to always check is to click on the link of the generated html. There were also 2 package names in the original that weren't correct, preventing the link from being generated properly. We'll be able to merge after these corrections.

The exercise-solution sections look good. Thanks for your work on this!
